public/main.js

```diff
@@ -4,6 +4,39 @@ var form = document.getElementById("todo-form");
 var todoTitle = document.getElementById("new-todo");
 var error = document.getElementById("error");
 var completeCounter = document.getElementById("count-label");
+var deleteComplete = document.getElementById("delete-complete");
+var all = document.getElementById("all");
+var active = document.getElementById("active");
+var completed = document.getElementById("completed");
+var displayMessage = document.getElementById("display-message");
+var messages = [
+    "Displaying all todo items.",
+    "Displaying only currently active todo items.",
+    "Displaying only completed todo items."
+];
+var ALL_VALUES = 0;
+var ACTIVE_VALUES = 1;
+var COMPLETE_VALUES = 2;
+var mode = ALL_VALUES;
+displayMessage.innerHTML = messages[ALL_VALUES];
+
+all.onclick = function() {
+    mode = ALL_VALUES;
+    displayMessage.innerHTML = messages[mode];
+    reloadTodoList();
+};
+
+active.onclick = function() {
+    mode = ACTIVE_VALUES;
+    displayMessage.innerHTML = messages[mode];
+    reloadTodoList();
+};
+
+completed.onclick = function() {
+    mode = COMPLETE_VALUES;
+    displayMessage.innerHTML = messages[mode];
+    reloadTodoList();
+};
```

Comment:
The onclick functions on all, active and completed are identical, except for the mode. Any way you can abstract this out to another function?